compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectPaths.java

```diff
@@ -53,6 +53,11 @@ public static Path packageRoot(Path filepath) throws ProjectException {
             throw new ProjectException("provided path is not a regular file");
         }
 
+        // Check if the file is inside a Ballerina package directory
+        if (findProjectRoot(filepath).isEmpty()) {
+            throw new ProjectException("provided file path does not belong to a Ballerina package");
```

Comment:
Wouldn't it be better if we include the file path that the project API was trying to lookup in this error message?

Reply:
Addressed in c4a17d7